### PR TITLE
Cdn dasd 3693

### PR DESCRIPTION
**Pull request by qamago**
Added cdn-profile and cdn-endpoint linked templates into das platform building blocks.

### CODE REVIEW

File: templates/storage-account-arm.json [hunk 1]
```diff
@@ -84,6 +84,10 @@
         "storageConnectionString": {
             "type": "string",
             "value": "[concat('DefaultEndpointsProtocol=https;AccountName=',parameters('storageAccountName'),';AccountKey=',listKeys(resourceId('Microsoft.Storage/storageAccounts',parameters('storageAccountName')), providers('Microsoft.Storage', 'storageAccounts').apiVersions[0]).keys[0].value, ';EndpointSuffix=core.windows.net')]"
+        },
+        "storageAccountName": {
+            "type": "string",
+            "value": "[resourceId('Microsoft.Storage/storageAccounts', parameters('storageAccountName'))]"
```
Comment (EwanNoble) [inline templates/storage-account-arm.json]: Delete this

File: templates/cdn-profile.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            "sku": {
+                "name": "[parameters('cdnSKU')]"
+            },
+            "properties": {},
+            "resources": [],
+            "dependsOn": []
+        }
+    ],
+    "outputs": {
+        "cdnProfileName": {
```
Comment (EwanNoble) [inline templates/cdn-profile.json]: Delete this

File: templates/cdn-profile.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            "type": "Microsoft.Cdn/profiles",
+            "apiVersion": "2017-10-12",
+            "location": "[resourceGroup().location]",
+            "tags": {},
+            "sku": {
+                "name": "[parameters('cdnSKU')]"
+            },
+            "properties": {},
+            "resources": [],
+            "dependsOn": []
```
Comment (EwanNoble) [inline templates/cdn-profile.json]: Remove the empty properties 'resources' and 'dependsOn'

File: templates/cdn-endpoint.json [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                "[resourceId('Microsoft.Cdn/profiles/endpoints',parameters('cdnProfileName') , parameters('cdnEndPointName'))]"
+            ]
+        }
+    ],
+    "outputs": {
+        "hostName": {
+            "type": "string",
+            "value": "[reference(parameters('cdnEndPointName')).hostname]"
+        },
+        "originHostHeader": {
```
Comment (EwanNoble) [inline templates/cdn-endpoint.json]: Delete this

File: templates/cdn-endpoint.json [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+            "properties": {
+                "hostName": "[parameters('customDomainName')]"
+            },
+            "dependsOn": [
+                "[resourceId('Microsoft.Cdn/profiles/endpoints',parameters('cdnProfileName') , parameters('cdnEndPointName'))]"
+            ]
+        }
+    ],
+    "outputs": {
+        "hostName": {
```
Comment (EwanNoble) [inline templates/cdn-endpoint.json]: Rename to endpointHostName

File: templates/cdn-endpoint.json [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+                    "text/html",
+                    "text/css",
+                    "text/javascript",
+                    "application/x-javascript",
+                    "application/javascript",
+                    "application/json",
+                    "application/xml"
+                ],
+                "isCompressionEnabled": true,
+                "isHttpAllowed": true,
```
Comment (EwanNoble) [inline templates/cdn-endpoint.json]: Should we be allowing http?

File: templates/cdn-endpoint.json [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+                "isHttpsAllowed": true,
+                "queryStringCachingBehavior": "IgnoreQueryString",
```
Comment (EwanNoble) [inline templates/cdn-endpoint.json]: This might be better as a parameter

File: templates/cdn-endpoint.json [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+        "originHostName": "[replace(replace(parameters('originHostName'),'https://',''),'/','')]"
+    },
+    "resources": [
+        {
+            "name": "[variables('fullEndpointName')]",
+            "type": "Microsoft.Cdn/profiles/endpoints",
+            "apiVersion": "2017-10-12",
+            "location": "[resourceGroup().location]",
+            "tags": {},
+            "dependsOn": [],
```
Comment (EwanNoble) [inline templates/cdn-endpoint.json]: Remove the empty properties 'tags' and 'dependsOn'

**Comment by EwanNoble** [comment]
Please squash the PR before merging

File: templates/storage-account-arm.json (hunk 1 above)
Comment (qamago) [reply templates/storage-account-arm.json]: done

File: templates/cdn-profile.json (hunk 2 above)
Comment (qamago) [reply templates/cdn-profile.json]: done

File: templates/cdn-profile.json (hunk 3 above)
Comment (qamago) [reply templates/cdn-profile.json]: done

File: templates/cdn-endpoint.json [hunk 9]
```diff
@@ -0,0 +1,99 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "cdnProfileName": {
+            "type": "string",
+            "metadata": {
+                "description": "Name of Content Delivery Name profile"
```
Comment (EwanNoble) [inline templates/cdn-endpoint.json]: Content Delivery Network

File: templates/cdn-endpoint.json [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+            "defaultValue": ""
+        },
+        "isHttpAllowed": {
+            "type": "bool",
+            "defaultValue": false
+        }
+    },
+    "variables": {
+        "fullEndpointName": "[concat(parameters('cdnProfileName'), '/', parameters('cdnEndpointName'))]",
+        "CustomDomainEnabled": "[greater(length(parameters('customDomainName')), 0)]",
```
Comment (EwanNoble) [inline templates/cdn-endpoint.json]: Use camel case for this: 'customDomainEnabled'

File: templates/template.json [hunk 11]
```diff
@@ -0,0 +1,184 @@
+{
```
Comment (EwanNoble) [inline templates/template.json]: Delete this file

File: templates/cdn-profile.json [hunk 12]
```diff
@@ -0,0 +1,37 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "cdnProfileName": {
+            "type": "string",
+            "metadata": {
+                "description": "Name of Content Delivery Name (CDN) profile"
```
Comment (EwanNoble) [inline templates/cdn-profile.json]: Content Delivery Network

File: templates/cdn-endpoint.json (hunk 9 above)
Comment (qamago) [reply templates/cdn-endpoint.json]: done

File: templates/cdn-endpoint.json (hunk 10 above)
Comment (qamago) [reply templates/cdn-endpoint.json]: done

File: templates/cdn-profile.json (hunk 12 above)
Comment (qamago) [reply templates/cdn-profile.json]: done

File: templates/cdn-endpoint.json (hunk 7 above)
Comment (qamago) [reply templates/cdn-endpoint.json]: done

**Pull request merged**